specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2019-05-01-preview/examples/CustomDomains_CreateOrUpdate.json

```diff
@@ -0,0 +1,31 @@
+{
+  "parameters": {
+    "domainResource": {
+      "properties": {
+        "thumbprint": "934367bf1c97033f877db0f15cb1b586957d3133",
+        "appName": "myapp",
+        "certName": "mycert"
+      }
+    },
+    "api-version": "2019-05-01-preview",
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "resourceGroupName": "myResourceGroup",
+    "serviceName": "myservice",
+    "appName": "myapp",
+    "domainName": "mydomain.com"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "properties": {
+          "thumbprint": "934367bf1c97033f877db0f15cb1b586957d3133",
+          "appName": "myapp",
+          "certName": "mycert"
+        },
+        "type": "Microsoft.AppPlatform/Spring/apps/domains",
+        "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.AppPlatform/Spring/myservice/apps/myapp/domains/mydomain.com",
```

Comment:
>mydomain.com [](start = 166, length = 12)

Resource names have limitations (documented at [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#arguments-for-crud-on-resource](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#arguments-for-crud-on-resource)). This may limit the domains you can support with this API. Is that going to be a problem?

Reply:
I think should be no problem based on domain names [spec](https://tools.ietf.org/html/rfc1035), which is totally under the restriction of ARM.
```
They must
start with a letter, end with a letter or digit, and have as interior
characters only letters, digits, and hyphen.

To simplify implementations, the total length of a domain name (i.e.,
label octets and label length octets) is restricted to 255 octets or
less.
```

Reply:
Great. Thanks for confirming.